src/main/webapp/routing.js

```diff
@@ -0,0 +1,170 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* exported calculateAndDisplayRoute hideRouteMarkers showRouteMarkers */
+/* globals casesData map */
+
+let chosenRoute = 0;
+let routeLines = [];
+let routeMarkers = [];
+
+/**
+ * Get route from Directions API
+ * Calulate cases for each route
+ * Display route with fewest cases
+ */
+function calculateAndDisplayRoute(directionsService, mapObject) {
+  directionsService.route(
+      {
+        origin: {query: document.getElementById('start').value},
+        destination: {query: document.getElementById('end').value},
+        travelMode: google.maps.TravelMode.DRIVING,
```

Comment:
How do users choose the source/destination? Is driving mode always the optimal choice?

Reply:
Source/destination have a text box input. Since the smallest area the data comes in is counties, I thought driving made the most sense although that could be changed later.

Reply:
It should be fine to keep this for now. But we have more things to resolve, for example, in what granularity can users choose locations? Will the users be able to choose the travel mode, e.g. from one country to another with biking? These can be refined and added to the feature list in our design doc. The overall goal is to build a product that's useful to target users, even with limited functionalities.